docs/modules/ROOT/pages/amazon-cognitouserpools.adoc

```diff
@@ -0,0 +1,276 @@
+= Amazon Cognito User Pools Client
+
+include::./includes/attributes.adoc[]
+
+Service client for accessing Amazon Cognito user pools (Identity Provider).
+
+Using the Amazon Cognito user pools API, you can create a user pool to manage directories and users.
+You can authenticate a user to obtain tokens related to user identity and access policies.
+
+For more information, see the https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools.html[Amazon Cognito user pools] (https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/cognitoidentityprovider/package-frame.html[AWS Java SDK 2.x docs]).
+
+
+NOTE: The Cognito User Pools extension is based on https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/welcome.html[AWS Java SDK 2.x].
+It's a major rewrite of the 1.x code base that offers two programming models (Blocking & Async).
+
+The Quarkus extension supports two programming models:
+
+* Blocking access using URL Connection HTTP client (by default) or the Apache HTTP Client
+* https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/basics-async.html[Asynchronous programming] based on JDK's `CompletableFuture` objects and the Netty HTTP client.
+
+In this guide, we see how you can get your REST services to use Cognito User Pool locally and on AWS.
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* JDK 11+ installed with `JAVA_HOME` configured appropriately
+* an IDE
+* Apache Maven {maven-version}
+* One of:
+** An AWS Account to access the Cognito service
+** Or Docker for your system to run Cognito locally for testing purposes
+
+=== Set up Cognito locally
+
+The easiest way to start working with Cognito is to run a local mock instance as a container.
+
+[source,bash,subs="verbatim,attributes"]
+----
+docker run --rm --name moto -p 5000:5000 -d motoserver/moto:3.0.2
+----
+This starts a Cognito mock instance that is accessible on port `5000`.
+
+Create an AWS profile for your local instance using AWS CLI:
+[source,shell,subs="verbatim,attributes"]
+----
+$ aws configure --profile moto
+AWS Access Key ID [None]: test-key
+AWS Secret Access Key [None]: test-secret
+Default region name [None]: us-east-1
+Default output format [None]: test
+----
+
+=== Create a Cognito User Pool
+
+Create a Cognito user pool using AWS CLI.
+
+[source,bash,subs="verbatim,attributes"]
+----
+aws --profile moto --endpoint-url=http://localhost:5000 cognito-idp create-user-pool --pool-name quarkus-users-dev
+----
+
+Or, if you want to use your Cognito user pool on your AWS account create a user pool using your default profile
+[source,bash,subs="verbatim,attributes"]
+----
+aws cognito-user-pools cognito-idp --pool-name quarkus-users-dev
+----
+
+== Solution
+The application built here allows count Cognito User Pools into current region.
```

Comment:
Odd grammar. Maybe `The application built here shows the count of Cognito User Pools in the current region.`